edge/pkg/edged/edged_pods.go

```diff
@@ -530,3 +532,331 @@ func (e *edged) removeOrphanedPodStatuses(pods []*v1.Pod) {
 
 	e.statusManager.RemoveOrphanedStatuses(podUIDs)
 }
+
+// GetPodCgroupParent gets pod cgroup parent from container manager.
+func (e *edged) GetPodCgroupParent(pod *v1.Pod) string {
+	/*pcm := e.containerManager.NewPodContainerManager()
+	_, cgroupParent := pcm.GetPodContainerName(pod)
+	return cgroupParent*/
+	return "systemd"
+}
+
+// GenerateRunContainerOptions generates the RunContainerOptions, which can be used by
+// the container runtime to set parameters for launching a container.
+func (e *edged) GenerateRunContainerOptions(pod *v1.Pod, container *v1.Container, podIP string) (*kubecontainer.RunContainerOptions, func(), error) {
+	/*opts, err := e.GenerateContainerOptions(pod)
+	if err != nil {
+		return nil, nil, err
+	}*/
+	opts := kubecontainer.RunContainerOptions{}
+
+	hostname, hostDomainName, err := e.GeneratePodHostNameAndDomain(pod)
+	if err != nil {
+		return nil, nil, err
+	}
+	opts.Hostname = hostname
+	podName := util.GetUniquePodName(pod)
+	volumes := e.volumeManager.GetMountedVolumesForPod(podName)
+	opts.PortMappings = kubecontainer.MakePortMappings(container)
+
+	// TODO: remove feature gate check after no longer needed
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		blkutil := volumepathhandler.NewBlockVolumePathHandler()
+		blkVolumes, err := e.makeBlockVolumes(pod, container, volumes, blkutil)
+		if err != nil {
+			return nil, nil, err
+		}
+		opts.Devices = append(opts.Devices, blkVolumes...)
+	}
+
+	/*envs, err := e.makeEnvironmentVariables(pod, container, podIP)
```

Comment:
I think we need support for env vars . Can we track via an issue and handle post 1.0 ?

Reply:
Ok will raise issue